core/modules/filters/eachday.js

```diff
@@ -24,9 +24,16 @@ exports.eachday = function(source,operator,options) {
 		value = (new Date(value)).setHours(0,0,0,0);
 		return value+0;
 	};
+	// Convert ISO date string to Date object if needed.
+	var toDateObj = function(value) {
+		if(!(value instanceof Date)) {
+			value = new Date($tw.utils.parseDate(value));
+		}
+		return value;
+	};
 	source(function(tiddler,title) {
 		if(tiddler && tiddler.fields[fieldName]) {
-			var value = toDate(tiddler.fields[fieldName]);
+			var value = toDate(toDateObj(tiddler.fields[fieldName]));
```

Comment:
I think we can remove the `toDateObj()` function and instead change this line to:

```
var value = toDate($tw.utils.parseDate(tiddler.fields[fieldName]));
```

The reason is that both `toDate()` and `isSameDay()` convert their argument to a date if necessary, and `$tw.utils.parseDate()` copes correctly with an argument that is already a date object.